src/extlibs/hrvo/visualizer/hrvo_visualizer.py

```diff
@@ -0,0 +1,181 @@
+
+    def toggle_pause(event):
+        nonlocal is_paused
+        if is_paused:
+            robot_anim.resume()
+        else:
+            robot_anim.pause()
+        is_paused = not is_paused
+
+    def on_plot_hover(event):
+        # Iterating over each data member plotted
+        for robot_id, line in enumerate(line_list):
+            # Searching which data member corresponds to current mouse position
+            if line.contains(event)[0]:
+                line.set_alpha(1.0)
+                robot_id_text.set_text(f"Robot {robot_id}")
+            else:
+                line.set_alpha(default_line_alpha)
+
+    # set up plot
+    fig = plt.figure()
+    fig.suptitle(file_name)
+    ax = fig.add_subplot(
+        111,
+        autoscale_on=False,
+        xlim=(min_robot_x_pos - x_offset, max_robot_x_pos + x_offset),
+        ylim=(min_robot_y_pos - y_offset, max_robot_y_pos + y_offset),
+    )
+    ax.set_aspect("equal")
+    ax.grid()
+    (line,) = ax.plot([], [], "--r")
+    # plt.subplots_adjust(left=0.3)
+    time_text = ax.text(0.5, 0.2, "", transform=plt.gcf().transFigure)
+    robot_id_text = ax.text(0.5, 0.1, "", transform=plt.gcf().transFigure)
+
+    # set-up robot circles
+    initial_frame = robot_pos_df[robot_pos_df["frame"] == 0]
+    robot_list = []
+    line_list = []
+    for robot_id in range(num_robots):
+        radius = float(
+            initial_frame[initial_frame["robot_id"] == robot_id]["radius"].head(1)
+        )
+        robot = Circle((0, 0), radius, facecolor="aqua", edgecolor="black")
+        robot_list.append(robot)
+        line_obj = ax.plot([], [], linestyle="--", alpha=default_line_alpha)[0]
+        line_list.append(line_obj)
+
+    # Start/Stop on click
+    fig.canvas.mpl_connect("button_press_event", toggle_pause)
+
+    # TODO: Allow highlighting after animation is over. Can achieve this by not passing frames to FunAnimation
+    # Highlight tracked line on hover
+    fig.canvas.mpl_connect("motion_notify_event", on_plot_hover)
+
+    # Animate
+    print(f"Animating robots from {file_location}")
+    robot_anim = FuncAnimation(
+        fig,
+        func=update,
+        interval=frame_length_ms / play_back_speed,
+        init_func=setup_plot,
+        frames=num_frames,
+        blit=True,
+        repeat=False,
+    )
+    plt.show()
+
+    if gif_output_file is not None:
+        # save gif
+        print("==============================================")
+        print(f"Saving gif of robots to {gif_output_file}...")
+        robot_anim.save(gif_output_file, writer=PillowWriter(fps=frame_rate))
+        print(f"Saved gif of robots to {gif_output_file}")
+
+
+file_name = "1_robot_moving_in_square"
+file_location = (
+    f"/home/nima/thunderbots/Software/src/software/hrvo/hrvo_data/{file_name}.csv"
+)
+df = pd.read_csv(file_location)  # , skiprows=3)
+animate_robots(
+    df
+)  # , f'/home/nima/thunderbots/Software/src/software/hrvo/hrvo_data/{file_name}.gif')
```

Comment:
can we make these paths non-Nima specific? (probably something passed in)

Reply:
Updated so the outputs are stored and read from `/tmp`